backend/src/model/entity/timeline.rs

```diff
@@ -0,0 +1,133 @@
+use chrono::NaiveDate;
+use diesel::{
+    sql_query,
+    sql_types::{BigInt, Date, Nullable},
+    QueryResult, QueryableByName,
+};
+use diesel_async::{AsyncPgConnection, RunQueryDsl};
+use std::collections::HashMap;
+
+use crate::model::dto::timeline::{TimelineDto, TimelineEntryDto, TimelineParameters};
+
+/// Stores the result of the timeline query. Dates and sum of addtions and removals.
+#[derive(QueryableByName, Debug)]
+struct TimelineQeueryResult {
+    #[diesel(sql_type = Nullable<Date>)]
+    date: Option<NaiveDate>,
+    #[diesel(sql_type = BigInt)]
+    additions: i64,
+    #[diesel(sql_type = BigInt)]
+    removals: i64,
+}
+
+/// Summarize all plantings into a timeline.
+///
+/// # Errors
+/// * Unknown, diesel doesn't say why it might error.
+pub async fn calculate(
+    params: TimelineParameters,
+    conn: &mut AsyncPgConnection,
+) -> QueryResult<TimelineDto> {
+    // the following query gives us additions and removals per day
+    // date        additions  removals
+    // 2000-01-01  7          3
+    // 2000-01-02  4          0
+    // 2000-01-03  2          8
+
+    let query = sql_query(
+        "SELECT
+            COALESCE(t1.add_date, t2.remove_date) as date,
+            COALESCE(t1.additions, 0) as additions,
+            COALESCE(t2.removals, 0) as removals
+        FROM
+            (
+                SELECT
+                    add_date,
+                    COUNT(*) as additions
+                FROM
+                    plantings
+                WHERE
+                    add_date BETWEEN $1 AND $2
+                GROUP BY
+                    add_date
+            ) t1
+        FULL OUTER JOIN
```

Comment:
@markus2330 
> I think it is a good idea to write it as raw sql. I assume it was much easier to write?

When should we use raw SQL? I guess the above can be expressed in the diesel dsl.
Full outer join can be expressed via left outer joins
See: https://en.wikipedia.org/wiki/Join_%28SQL%29#Full_outer_join


I definitely would want a automated test for that.
See: https://docs.diesel.rs/master/diesel/fn.sql_query.html#safety

Reply:
> When should we use raw SQL?

My proposal last week was: when it becomes too time consuming to express it via Diesel.
Last time @horenso said it might not be necessary at all but maybe we have a counter example here.
We will discuss it next meeting.

> Full outer join can be expressed via left outer joins

Maybe we use as guideline: we use raw SQL whenever we cannot directly express the SQL we want in Diesel.

> I definitely would want a automated test for that.

Yes, I agree. We definitely need tests for the timeline endpoint.

Reply:
> Also there is no support for OUTER JOIN in diesel at the moment.

There is. Outer left join is supported. And outer right join is just outer left join with the operants reversed.